mypy/report.py

```diff
@@ -5,19 +5,21 @@
 import os
 import shutil
 
-from typing import Callable, Dict, List, cast
+from typing import Callable, Dict, List, Tuple, cast
 
-from mypy.types import Type
-from mypy.nodes import MypyFile, Node
+import mypy.build
+from mypy.nodes import MypyFile, Node, FuncDef
 from mypy import stats
+from mypy.traverser import TraverserVisitor
+from mypy.types import Type
 
 
 reporter_classes = {}  # type: Dict[str, Callable[[Reports, str], AbstractReporter]]
 
 
 class Reports:
-    def __init__(self, main_file: str, data_dir: str, report_dirs: Dict[str, str]) -> None:
-        self.main_file = main_file
+    def __init__(self, sources: List['mypy.build.BuildSource'], data_dir: str, report_dirs: Dict[str, str]) -> None:
```

Comment:
OK, I cut this logic entirely (it was kind of silly legacy logic in any case), which eliminates what's left of the import cycle.